pkgs/test_core/lib/src/runner/configuration.dart

```diff
@@ -41,61 +41,61 @@ class Configuration {
 
   /// Whether `--help` was passed.
   bool get help => _help ?? false;
-  final bool _help;
+  final bool? _help;
 
   /// Custom HTML template file.
-  final String customHtmlTemplatePath;
+  final String? customHtmlTemplatePath;
 
   /// Whether `--version` was passed.
   bool get version => _version ?? false;
-  final bool _version;
+  final bool? _version;
 
   /// Whether to pause for debugging after loading each test suite.
   bool get pauseAfterLoad => _pauseAfterLoad ?? false;
-  final bool _pauseAfterLoad;
+  final bool? _pauseAfterLoad;
 
   /// Whether to run browsers in their respective debug modes
   bool get debug => pauseAfterLoad || (_debug ?? false) || _coverage != null;
-  final bool _debug;
+  final bool? _debug;
 
   /// The output folder for coverage gathering
-  String get coverage => _coverage;
-  final String _coverage;
+  String? get coverage => _coverage;
+  final String? _coverage;
 
   /// The path to the file from which to load more configuration information.
   ///
   /// This is *not* resolved automatically.
   String get configurationPath => _configurationPath ?? 'dart_test.yaml';
-  final String _configurationPath;
+  final String? _configurationPath;
 
   /// The path to dart2js.
   String get dart2jsPath => _dart2jsPath ?? p.join(sdkDir, 'bin', 'dart2js');
-  final String _dart2jsPath;
+  final String? _dart2jsPath;
 
   /// The name of the reporter to use to display results.
   String get reporter => _reporter ?? defaultReporter;
-  final String _reporter;
+  final String? _reporter;
```

Comment:
Can these somehow use the `late` pattern so that they may be non-nullable?

Reply:
That's the pattern I'm leaning towards and it has seemed appropriate so far. The other use is using a variable that needs to be passed in it's initializer, like what happens with stream subscriptions.

Reply:
Ah right I have used it that way as well

Reply:
I wonder where best practices for this are being documented? Should we update effective dart?

Reply:
cc @munificent

Reply:
We should update Effective Dart at some point, but I want to make sure we have enough time for the best practices to settle out before we start telling people what to do.